eclipse/tests/src/nl/tudelft/watchdog/eclipse/logic/ui/listeners/MarkupModelListenerTest.java

```diff
@@ -0,0 +1,251 @@
+import nl.tudelft.watchdog.core.logic.event.eventtypes.TrackingEventType;
+import nl.tudelft.watchdog.core.logic.event.eventtypes.staticanalysis.StaticAnalysisWarningEvent;
+import nl.tudelft.watchdog.core.logic.ui.events.WatchDogEventType;
+import nl.tudelft.watchdog.eclipse.logic.interval.IntervalManager;
+import nl.tudelft.watchdog.eclipse.logic.network.TransferManager;
+import nl.tudelft.watchdog.eclipse.logic.ui.listeners.WorkbenchListener;
+
+import static org.junit.Assert.assertTrue;
+
+public class MarkupModelListenerTest {
+
+	private WorkbenchListener workbenchListener;
+	private TrackingEventManager trackingEventManager;
+	private TransferManager transferManager;
+	private IWorkspace workspace;
+	private IProject project;
+	private IFile testFile;
+	private IFile preExistingTestFile;
+	private IMarker preExistingMarker;
+
+	@Before
+	public void setup() throws Exception {
+		WatchDogEventType.intervalManager = Mockito.mock(IntervalManager.class);
+		this.trackingEventManager = Mockito.mock(TrackingEventManager.class);
+		this.transferManager = Mockito.mock(TransferManager.class);
+
+		this.setUpTestingProject();
+
+		this.workbenchListener = new WorkbenchListener(trackingEventManager, transferManager);
+		this.workbenchListener.attachListeners();
+	}
+
+	private void setUpTestingProject() throws Exception {
+		this.workspace = ResourcesPlugin.getWorkspace();
+		IWorkspaceDescription description = this.workspace.getDescription();
+		description.setAutoBuilding(true);
+		this.workspace.setDescription(description);
+
+		this.project = this.workspace.getRoot().getProject("Testing");
+		this.project.create(this.workspace.newProjectDescription(project.getName()), null);
+		this.project.open(null);
+
+		IFolder folder = project.getFolder("src");
+		folder.create(true, true, null);
+
+		this.testFile = folder.getFile("Main.java");
+		this.testFile.create(generateFileStreamLines(25), true, null);
+
+		this.preExistingTestFile = folder.getFile("Existing.java");
+		this.preExistingTestFile.create(generateFileStreamLines(25), true, null);
+		this.preExistingMarker = this.preExistingTestFile.createMarker(IMarker.PROBLEM);
+		this.preExistingTestFile.createMarker(IMarker.PROBLEM);
+
+		this.workspace.save(true, null);
+
+		Job.getJobManager().join(ResourcesPlugin.FAMILY_AUTO_BUILD, null);
+	}
+
+	private InputStream generateFileStreamLines(int numLines) {
+		StringBuilder builder = new StringBuilder();
+
+		for (int i = 0; i < numLines; i++) {
+			builder.append("Line ")
+				   .append(i)
+				   .append('\n');
+		}
+
+		return new ByteArrayInputStream(builder.toString().getBytes(StandardCharsets.UTF_8));
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		this.project.delete(true, true, null);
+		this.workspace.save(true, null);
+
+		Job.getJobManager().join(ResourcesPlugin.FAMILY_AUTO_BUILD, null);
+		this.workbenchListener.shutDown();
+	}
+
+	@Test
+	public void picks_up_created_warnings_on_first_save() throws Exception {
```

Comment:
These method names do not exactly follow camel casing

Reply:
This is a personal preference. We use this casing in Mockito. You want to change it?